src/commands/update-github-release.ts

```diff
@@ -0,0 +1,135 @@
+import * as Octokit from '@octokit/rest';
+import * as yargsParser from 'yargs-parser';
+import { getCurrentRepoNameWithOwner } from '../git/git';
+
+const BADGE = '[update-github-release]\t';
+
+type GitTag = string;
+type GitHubRepo = {
+  owner: string;
+  name: string;
+};
+
+export async function run(...args): Promise<boolean> {
+  const argv = yargsParser(args);
+
+  console.log(`${BADGE}`, argv);
+
+  const success = await updateGitHubRelease(argv.versionTag, argv.title, argv.text, !!argv.dry);
+
+  if (success) {
+    console.log(`${BADGE}Success.`);
+  }
+
+  return success;
+}
+
+/**
+ * Updates the corresponding GitHub release for `versionTag` using the given `title` and `text`.
+ */
+export async function updateGitHubRelease(
+  versionTag: GitTag,
+  title: string,
+  text: string,
+  dryRun: boolean = false
+): Promise<boolean> {
+  const repo = getCurrentRepoNameWithOwnerAsObject(getCurrentRepoNameWithOwner());
+  const octokit = await createOctokit(process.env.GH_TOKEN);
+  const releaseId = await getExistingReleaseId(octokit, repo, versionTag);
+  const releaseExists = releaseId != null;
+
+  if (releaseExists) {
+    if (dryRun) {
+      console.log(`${BADGE}Would now update existing release. Skipping since this is a dry run!`);
+
+      return true;
+    }
+
+    console.log(`${BADGE}Updating existing release for ${versionTag} ...`);
+
+    return updateExistingRelease(octokit, repo, releaseId, title, text);
+  }
+
+  if (dryRun) {
+    console.log(`${BADGE}Would now create a new release. Skipping since this is a dry run!`);
+
+    return true;
+  }
+
+  console.log(`${BADGE}Creating new release for ${versionTag} ...`);
+
+  return createNewRelease(octokit, repo, versionTag, title, text);
+}
+
+async function updateExistingRelease(
+  octokit: Octokit,
+  repo: GitHubRepo,
+  releaseId: number,
+  title: string,
+  text: string
+): Promise<boolean> {
+  const response = await octokit.repos.editRelease({
+    owner: repo.owner,
+    repo: repo.name,
+    release_id: releaseId,
+    name: title,
+    body: text
+  });
+
+  return response.status === 200;
+}
+
+async function createNewRelease(
+  octokit: Octokit,
+  repo: GitHubRepo,
+  versionTag: GitTag,
+  title: string,
+  text: string
+): Promise<boolean> {
+  const isPrerelease = !!versionTag.match(/-/);
```

Comment:
Bin kein Fan dieser "!!" notation. 
Da gibts zu viele Pitfalls und es macht den Code obskurer. 

Ich kann hier z.B. nicht erkennen, ob `versionTag.match` bei keinem Match "false" oder "undefined" zurück gibt - was ein wichtiger Unterschied ist. 

Daher würde ich derartige Bedingungen immer explizit formulieren (entspricht ja auch einemd er 5Minds Credos 😉 )

Ich nehme jetzt einfach mal an, dass bei einem match "true" zurück kommt. 
Falls das nicht stimmt, müsste das Snippet angepasst werden:

```suggestion
  const isPrerelease = versionTag.match(/-/) === true;
```

Siehe z.B.
https://www.sitepoint.com/javascript-double-negation-trick-trouble/

Reply:
Ich kommentiere mal hier zu beidem: Habe die Zeile jetzt geändert, bin aber bzgl. des Sentiments anderer Meinung. Ich persönlich halte den verlinkten Artikel für sehr schwach. Dass er "unary plus" und Performance-Erwägungen in einen Artikel zu "Du musst wissen, wie Truthyness in JS funktioniert" mischt, spricht nicht für die Autoren.

Wie dem auch sei: Wir prüfen jetzt den `match` per `!= null`. 

Als Anmerkung zum anderen Kommentar weiter unten: `null/undefined` sollte generell nicht  `===` bzw.  `!==` mit geprüft werden. Die Unterscheidung zwischen `null` und `undefined` ist eine akademische, in unserem Fall willkürliche und die von uns verwendeten Bibliotheken und Pakete behandeln diese Werte nicht auf die gleiche Art, wie wir sie gerne verstehen wollen. Das führt zu Fehlern, wenn man mit Code von anderen interagiert.

Da `x == null` sowohl für `x = null` als auch für `x = undefined` wahr ist, sollte jeder Programmierer, der an funktionierendem, robusten Code interessiert ist, erkennen, dass dies die eine Stelle ist, an der "type inference" gewünscht ist (bei anderen Vergleichen ist `===` natürlich zu bevorzugen).